entity-cluster-bot/README.md

```diff
@@ -12,7 +12,104 @@ Further, contracts created by an address are also considered part of the same en
 
 The bot generates a graph and identifies and reports on the connected components.
 
-A entity is created if this condition is observed accounts with less than MAX_NONCE transactions within a ENTITY_CLUSTER_WINDOW_IN_DAYS timeframe. Entities will age out within MAX_AGE_IN_DAYS.
+A entity is created if this condition is observed accounts with less than MAX_NONCE transactions. Entities will age out within MAX_AGE_IN_DAYS.
+
+
+## Sharding Implementation Details
+The bot has a graph of the connections between the addresses that must be shared between many intances. All instances read and write the shared graph at the same time, so to avoid race condition it use a 
+mutex implemented in dynamodb using  [DynamoDb conditional write](http://docs.aws.amazon.com/amazondynamodb/latest/developerguide/WorkingWithItems.html#WorkingWithItems.ConditionalUpdate) and [atomic compare-and-swap](https://en.wikipedia.org/wiki/Compare-and-swap) 
+
+Also we store the graph in s3 as the graph compressed can be around 15MB and optimize costs
+
+Each instance updates the shared graoh every TX_SAVE_STEP 
+
+
+
+## Infrastructure
```

Comment:
@stevenlanders , is this sufficient information for you to setup the appropriate AWS infra pieces?